src/System.Private.DataContractSerialization/src/System/Xml/StringHandle.cs

```diff
@@ -227,15 +228,17 @@ private bool Equals2(int offset2, int length2, XmlBufferReader bufferReader2)
             return GetString() == _bufferReader.GetString(offset2, length2);
         }
 
-        private bool Equals2(StringHandle s2)
+        public bool Equals(StringHandle other)
         {
-            StringHandleType type = s2._type;
+            if (ReferenceEquals(other, null))
```

Comment:
This has me wondering: Should we short-circuit here on positive reference equality (`if (ReferenceEquals(this, other) return true;`), particularly since it looks like we're about to do some O(n) comparisons (and possibly even some byte\[\] copies)?